### PR TITLE
Update notebook contrib guidance, update a few notebooks for site

### PR DESCRIPTION
**Pull request by jackgerrits**
## Why are these changes needed?

Uses \```{=mdx} blocks so fewer \\ are needed

## Related issue number

#1597 

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

**Review by ekzhu — APPROVED** [review]
There are some syntax changes in the notebook's code blocks. Can you run each of the notebooks locally just to sanity check if they work?

File: notebook/agentchat_function_call_async.ipynb [hunk 1]
```diff
@@ -44,6 +44,7 @@
    "outputs": [],
    "source": [
     "import time\n",
+    "import os\n",
```
Comment (sonichi) [inline notebook/agentchat_function_call_async.ipynb:47]: Is this change needed?

Comment (jackgerrits) [reply notebook/agentchat_function_call_async.ipynb]: its not, removed
